packages/nimble/inst/tests/test-coreR.R

```diff
@@ -726,3 +726,81 @@ seqTestsResults <- lapply(seqTests, test_coreRfeature)
 nonSeqIndexTestsResults <- lapply(nonSeqIndexTests, test_coreRfeature)
 indexChainTestsResults <- lapply(indexChainTests, test_coreRfeature)
 logicalTestsResults <- lapply(logicalTests, test_coreRfeature)
+
+## Some tests of using coreR features in BUGS models
+
+mc <- nimbleCode({
+    a ~ dnorm(0,1)
+    b[1:2] <- c(a, 1.1)
+})
+
+m <- nimbleModel(mc, inits = list(a = 2))
+cm <- try(compileNimble(m))
+test_that('c(a, 1.1) in BUGS compiles',
+          expect_false(inherits(cm, 'try-error')))
+test_that('c(a, 1.1) in BUGS',
+          expect_identical(as.numeric(m$b), c(2, 1.1)))
```

Comment:
Is it worth strengthening your first `test_that(...)` to also check compiled behavior?
```
test_that('c(a, 1.1) in compiled BUGS', {
    cm <- compileNimble(m)
    expect_identical(as.numeric(cm$b), c(2, 1.1))
})